hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -509,6 +510,14 @@ public int hashCode() {
     return uuid.hashCode();
   }
 
+  // Skip The OpStates which may change in Runtime.
+  public int getSignature() {
+    return Objects
+        .hash(uuid, uuidString, ipAddress, hostName, ports,
+            certSerialId, version, setupTime, revision, buildDate,
```

Comment:
I think it will be good to include this information here, we can show the updated detail information like version, build date on the UI to the users?